users/service.go

```diff
@@ -69,7 +70,7 @@ func New(users UserRepository, hasher Hasher, idp IdentityProvider) Service {
 func (svc usersService) Register(ctx context.Context, user User) error {
```

Comment:
I recommend that all methods return our `Error`. You don't want to do ugly casting every time you want to do something with it.

Reply:
I make changes in `Login` method to return `errors.Error` and in all other code to make that possible. Can you please review these changes before I continue with other methods